docker/0001-PyQt5-Raspberry-Pi-Build.patch

```diff
@@ -0,0 +1,72 @@
+Date: Wed, 10 Jul 2019 19:26:06 +0000
+Subject: [PATCH] PyQt5 Raspberry Pi Build
+
+---
+ build_scripts/platforms/linux.py             | 1 +
+ build_scripts/utils.py                       | 2 +-
+ sources/pyside2/PySide2/CMakeLists.txt       | 3 +++
+ sources/pyside2/PySide2/QtGui/CMakeLists.txt | 2 ++
+ 4 files changed, 7 insertions(+), 1 deletion(-)
+
+diff --git a/build_scripts/platforms/linux.py b/build_scripts/platforms/linux.py
+index 067179cd..91e831e8 100644
+--- a/build_scripts/platforms/linux.py
++++ b/build_scripts/platforms/linux.py
+@@ -66,6 +66,7 @@ def prepare_standalone_package_linux(self, vars):
+     if constrain_modules:
+         accepted_modules = ["libQt5" + module + "*.so.?" for module in constrain_modules]
+     accepted_modules.append("libicu*.so.??")
++    accepted_modules.append("libharfbuzz.so")
+ 
+     copydir("{qt_lib_dir}", destination_lib_dir,
+             filter=accepted_modules,
+diff --git a/build_scripts/utils.py b/build_scripts/utils.py
+index b34d3712..0c195246 100644
+--- a/build_scripts/utils.py
++++ b/build_scripts/utils.py
+@@ -849,7 +849,7 @@ def ldd(executable_path):
+     # List of ld's considered by ldd on Ubuntu (here's hoping it's the
+     # same on all distros).
+     rtld_list = ["/lib/ld-linux.so.2", "/lib64/ld-linux-x86-64.so.2",
+-        "/libx32/ld-linux-x32.so.2"]
++        "/libx32/ld-linux-x32.so.2", "/lib/ld-linux-armhf.so.3"]
+ 
+     # Choose appropriate runtime dynamic linker.
+     for rtld in rtld_list:
+diff --git a/sources/pyside2/PySide2/CMakeLists.txt b/sources/pyside2/PySide2/CMakeLists.txt
+index d666751e..34f8e91f 100644
+--- a/sources/pyside2/PySide2/CMakeLists.txt
++++ b/sources/pyside2/PySide2/CMakeLists.txt
+@@ -38,6 +38,9 @@ endif()
+ configure_file("${CMAKE_CURRENT_SOURCE_DIR}/support/__init__.py"
+                "${CMAKE_CURRENT_BINARY_DIR}/support/__init__.py" COPYONLY)
+ 
++configure_file("/usr/local/lib/libharfbuzz.so"
++               "${CMAKE_CURRENT_BINARY_DIR}" COPYONLY)
++
+ # now compile all modules.
+ file(READ "${CMAKE_CURRENT_BINARY_DIR}/pyside2_global.h" pyside2_global_contents)
+ 
+diff --git a/sources/pyside2/PySide2/QtGui/CMakeLists.txt b/sources/pyside2/PySide2/QtGui/CMakeLists.txt
+index 18d80f64..cfa5e6be 100644
+--- a/sources/pyside2/PySide2/QtGui/CMakeLists.txt
++++ b/sources/pyside2/PySide2/QtGui/CMakeLists.txt
+@@ -239,6 +239,7 @@ set(QtGui_include_dirs  ${QtGui_SOURCE_DIR}
+                         )
+ set(QtGui_libraries   pyside2
+                       ${Qt5Gui_LIBRARIES})
++	      #${Qt5Gui_LIBRARIES} /usr/local/lib/libharfbuzz.so)
+ set(QtGui_deps QtCore)
+ 
+ create_pyside_module(NAME QtGui
+@@ -252,4 +253,5 @@ create_pyside_module(NAME QtGui
+                      DROPPED_ENTRIES QtGui_DROPPED_ENTRIES)
+ 
+ install(FILES ${pyside2_SOURCE_DIR}/qpytextobject.h DESTINATION include/PySide2/QtGui/)
++#install(FILES /usr/local/lib/libharfbuzz.so DESTINATION lib/python3.5/site-packages/PySide2)
+ 
+-- 
+2.11.0
```

Comment:
Thanks... was hoping we could do more with this at some point... this allows the console to run on RPi, but will probably only be usable on the RPi 4 because of performance issues.